core/src/OAuth2/OAuth2Client.cs

```diff
@@ -98,7 +98,7 @@ public void AddBodyParameter(string key, string value)
             {
                 if (method == HttpMethod.Post)
                 {
-                    response = await HttpRequest.SendPost(endpointUri, _headers, _bodyParameters, requestContext);
+                    response = await HttpRequest.SendPost(endpointUri, _headers, new FormUrlEncodedContent(_bodyParameters), requestContext);
```

Comment:
>FormUrlEncodedContent [](start = 85, length = 21)

I believe in other places you removed the formurlencode from posts... is this needed?

Reply:
OK this change is not strictly necessary now. I'm reverting back to its original form. #ConsiderItDone